File: 83. Remove Duplicates from Sorted List/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
  public ListNode deleteDuplicates(ListNode head) {
    if (head == null){
      return head;
    }
    ListNode previous = head;
    ListNode current = head.next;
    while(current != null) {
      if (current.val != previous.val){
        previous.next = current;
        previous = current;
      }
      current = current.next;
    }
    previous.next = null;
    return head;
  }
}

File: 83. Remove Duplicates from Sorted List/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
  def deleteDuplicates(self, head):
    """
    :type head: ListNode
    :rtype: ListNode
    """
    if head == None:
      return head
    previous = head
    current = head.next
    while current != None:
      if current.val != previous.val:
        previous.next = current
        previous = current
      current = current.next
    previous.next = None
    return head